src/network.c

```diff
@@ -2684,6 +2724,54 @@ static int network_config_set_interface(const oconfig_item_t *ci, /* {{{ */
   return 0;
 } /* }}} int network_config_set_interface */
 
+static int
+network_config_set_bind_address(const oconfig_item_t *ci,
+                                struct sockaddr_storage **bind_address) {
+  if ((*bind_address) != NULL) {
+    ERROR("network_plugin: only a single bind address is allowed");
+    return -1;
+  }
+
+  char addr_text[256];
+
+  if (cf_util_get_string_buffer(ci, addr_text, sizeof(addr_text)) != 0)
+    return -1;
+
+  int ret;
+  struct addrinfo *res = NULL;
+  struct addrinfo hint = {.ai_family = PF_UNSPEC,
+                          .ai_flags = AI_NUMERICHOST,
+                          .ai_protocol = IPPROTO_IP,
+                          .ai_socktype = SOCK_DGRAM};
+
+  ret = getaddrinfo(addr_text, NULL, &hint, &res);
+  if (ret) {
+    ERROR("network plugin: Bind address option has invalid address set: %s",
+          gai_strerror(ret));
+    freeaddrinfo(res);
```

Comment:
I'm not sure this is correct. Posix doesn't say anything about what res points to on failure (http://pubs.opengroup.org/onlinepubs/9699919799/functions/freeaddrinfo.html). That means res can still be NULL, and I don't think you can call `freeaddrinfo` with a NULL pointer. On the other hand, there's no guarantee that res is NULL on failure and you don't have to freeaddrinfo on it.

Let's just do what the rest of the code in our codebase does and remove this call to `freeaddrinfo`.